spec/unit/serializer_spec.rb

```diff
@@ -0,0 +1,214 @@
+require "spec_helper"
+
+RSpec.describe IknowParams::Serializer do
+  shared_examples "a serializer" do
+    it "can load its type from a string" do
+      valid_values.zip(parsed_values).each do |valid, parsed|
+        expect(described_class.load(valid)).to eq parsed
+      end
+    end
+
+    it "raises an error when asked to load a bad value" do
+      invalid_values.each do |invalid|
+        expect { described_class.load(invalid) }.to(raise_error(Exception), "value was accepted: #{invalid}")
```

Comment:
I forgot to update this, and indeed, several serializers aren't obeying this interface. I'll make those changes.